projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,295 @@
+    lawsuit = lawsuit.replace("&lt;currentDate&gt;", DATE_FORMATTER.format(new Date()));
+
+    // User fields
+    lawsuit = lawsuit.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastName);
+    lawsuit = lawsuit.replace("&lt;userId&gt;", userID);
+    lawsuit = lawsuit.replace("&lt;userAddress&gt;", userAddress);
+    lawsuit = lawsuit.replace("&lt;userPhone&gt;", userPhone);
+    lawsuit = lawsuit.replace("&lt;userFax&gt;", userFax);
+
+    // First spam company
+    lawsuit = lawsuit.replace("&lt;companyName&gt;", companyName);
+    lawsuit = lawsuit.replace("&lt;companyId&gt;", companyId);
+    lawsuit = lawsuit.replace("&lt;companyAddress&gt;", companyAddress);
+    lawsuit = lawsuit.replace("&lt;companyPhone&gt;", companyPhone);
+    lawsuit = lawsuit.replace("&lt;companyFax&gt;", companyFax);
+
+    // Second spam company
+    lawsuit = lawsuit.replace("&lt;company2Name&gt;", company2Name);
+    lawsuit = lawsuit.replace("&lt;company2Id&gt;", company2Id);
+    lawsuit = lawsuit.replace("&lt;company2Address&gt;", company2Address);
+    lawsuit = lawsuit.replace("&lt;company2Phone&gt;", company2Phone);
+    lawsuit = lawsuit.replace("&lt;company2Fax&gt;", company2Fax);
+
+    return lawsuit;
+  }
+
+  private String readTemplate() {
+
+    try {
+      return new String(
+          Files.readAllBytes(Paths.get(lawsuitTemplateFilePath)), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      Log.w(TAG, "Error reading lawsuit's template.");
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  private String getPdfFilename() {
+
+    return DATE_TIME_FORMATTER.format(new Date()) + ".pdf";
+  }
+
+  private void initDirStructure() {
+
+    lawsuitMainPath =
+        Paths.get(
+                Environment.getExternalStorageDirectory().getPath(),
+                getString(R.string.app_name),
+                LAWSUIT_MAIN_DIR_NAME)
+            .toString();
+    makeDir(lawsuitMainPath);
+
+    lawsuitTemplateDirPath = Paths.get(lawsuitMainPath, LAWSUIT_TEMPLATE_DIR_NAME).toString();
+    makeDir(lawsuitTemplateDirPath);
+
+    // TODO: Template file stored at device storage. Perhaps move it to App resources.
+    try {
+      lawsuitTemplateFilePath =
+          Paths.get(lawsuitTemplateDirPath, LAWSUIT_TEMPLATE_FILE_NAME).toString();
+    } catch (Exception e) {
+      Toast.makeText(this, "No template file found for the lawsuit.", Toast.LENGTH_LONG).show();
+    }
+
+    // Output
+    lawsuitOutputPath = Paths.get(lawsuitMainPath, LAWSUIT_OUTPUT_DIR_NAME).toString();
+    makeDir(lawsuitOutputPath);
+  }
+
+  private void makeDir(String path) {
+
+    File directory = new File(path);
+    if (directory.exists() && directory.isDirectory()) directory.mkdirs();
+  }
+
+  private void initTimeZones() {
+
+    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+  }
+
+  private void getWriteStoragePermission() {
```

Comment:
Since this method calls `createPdf`, please make it clear from the method name.
Something like: `checkPermissionsThenCreatePdf()`.

Reply:
Changed it :)